src/types/token.rs

```diff
@@ -45,6 +45,17 @@ impl Token {
         }
     }
 
+    pub fn new_invalid() -> Self {
```

Comment:
I'm a little confused by this name. What's the intention of this function?

Reply:
The intention of this function is to create a new instance of `Token` that is intentionally invalid, so the `DatabaseContext` it's paired with can only be used for things that don't require a token.

Reply:
I’d prefer that we don’t rely on special values. Instead, let’s use an optional